### PR TITLE
ly: 0.2.1 -> 0.6.0

### PR DESCRIPTION
**Pull request by arjan-s**
###### Description of changes
Package upgrade, with changes required for building taken from https://github.com/NixOS/nixpkgs/pull/117646

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/display-managers/ly/default.nix [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
+  version = "0.6.0";
 
   src = fetchFromGitHub {
-    owner = "cylgom";
+    owner = "fairyglade";
     repo = "ly";
-    rev = version;
-    sha256 = "16gjcrd4a6i4x8q8iwlgdildm7cpdsja8z22pf2izdm6rwfki97d";
+    rev = "v${version}";
+    sha256 = "1vpcdswdr50qnpphvfywck6ksxb7qmljbml4hjy0hsdx6blc7igg";
```
Comment (drupol) [inline pkgs/applications/display-managers/ly/default.nix:11]: Use `hash` in place of `sha256`.

File: pkgs/applications/display-managers/ly/default.nix [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
     fetchSubmodules = true;
   };
 
-  buildInputs = [ linux-pam ];
-  makeFlags = [ "FLAGS=-Wno-error" ];
+  hardeningDisable = [ "all" ];
+  nativeBuildInputs = [ git ];
+  buildInputs = [ libxcb linux-pam ];
```
Comment (drupol) [inline pkgs/applications/display-managers/ly/default.nix:17]: Shouldn't `libxcb` be in `nativeBuildInputs` ?

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/applications/display-managers/ly/default.nix (hunk 2 above)
Comment (arjan-s) [reply pkgs/applications/display-managers/ly/default.nix:17]: As far as I understand it, `native*` contain packages from which binaries need to be run during the respective phases. That's not the case with `libxcb`, it's just a library that `ly` compiles against.

**Review by drupol — APPROVED** [review]
Thanks !

**Comment by drupol** [comment]
@GrahamcOfBorg build ly

**Comment by drupol** [comment]
@GrahamcOfBorg build ly

**Comment by arjan-s** [comment]
Result of `nixpkgs-review pr 237912` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ly</li>
  </ul>
</details>

File: pkgs/applications/display-managers/ly/default.nix [hunk 3]
```diff
@@ -23,7 +24,7 @@ stdenv.mkDerivation rec {
   meta = with lib; {
```
Comment (drupol) [inline pkgs/applications/display-managers/ly/default.nix:24]: Do you think we could add the `platforms` meta attribute?

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/applications/display-managers/ly/default.nix (hunk 3 above)
Comment (drupol) [reply pkgs/applications/display-managers/ly/default.nix:24]: Depends on which arch the derivation is working with.

Comment (arjan-s) [reply pkgs/applications/display-managers/ly/default.nix:24]: Thanks, without `platforms` it nixpkgs-review didn't even start to build on `aarch64-linux`, but now it does (successfully).

Comment (drupol) [reply pkgs/applications/display-managers/ly/default.nix:24]: This is not only for that, but also to display the proper platforms it runs on on search.nixos.org ;)

Comment (arjan-s) [reply pkgs/applications/display-managers/ly/default.nix:24]: According to https://github.com/fairyglade/ly it should work on "Linux and BSD", so I think it's good now.

**Comment by drupol** [comment]
@GrahamcOfBorg build ly

**Comment by patrickaldis** [comment]
Hi, just wondering what the current procedure to enable the display manager is? thanks for any help

**Comment by deviantsemicolon** [comment]
Will this ever be made into a module so I can enable it properly?

**Comment by Vonfry** [comment]
@deviantsemicolon  I create a draft pr #297434 to add a ly module, but it is not tested.

**Pull request merged**